tools/devops/automation/build-pipeline.yml

```diff
@@ -0,0 +1,288 @@
+  - checkout: self          # https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema#checkout
+    clean: true             # Executes: git clean -ffdx && git reset --hard HEAD
+    submodules: recursive
+
+  - checkout: maccore
+    clean: true
+    persistCredentials: true  # hugely important, else there are some scripts that check a single file from maccore that will fail
+
+  - checkout: templates
+    clean: true
+
+  - checkout: release-scripts
+    clean: true
+
+  - powershell: |
+      gci env: | format-table -autosize -wrap
+    displayName: 'Dump Environment'
+
+  - bash: |
+      security set-key-partition-list -S apple-tool:,apple: -s -k $OSX_KEYCHAIN_PASS login.keychain
+    env:
+      OSX_KEYCHAIN_PASS: $(xma-password) 
+    displayName: 'Remove security UI-prompt (http://stackoverflow.com/a/40039594/183422)'
+    condition: succeededOrFailed() # we do not care about the previous process cleanup
+
+  - task: xamops.azdevex.provisionator-task.provisionator@1
+    displayName: 'Provision Brew components'
+    inputs:
+      provisioning_script: $(Build.SourcesDirectory)/xamarin-macios/tools/devops/provision-brew-packages.csx
+      provisioning_extra_args: '-vvvv'
+      github-pat: $(GitHub.Token)
+    timeoutInMinutes: 30
+
+  - task: xamops.azdevex.provisionator-task.provisionator@1
+    displayName: 'Provision Python components'
+    inputs:
+      provisioning_script: $(Build.SourcesDirectory)/xamarin-macios/tools/devops/provision-python-packages.csx
+      provisioning_extra_args: '-vvvv'
+      github-pat: $(GitHub.Token)
+    timeoutInMinutes: 30
+
+  - bash: |
+      make -C $(Build.SourcesDirectory)/xamarin-macios/tools/devops build-provisioning.csx
+    displayName: 'Generate provisionator files.'
+
+  - task: xamops.azdevex.provisionator-task.provisionator@1
+    displayName: 'Provision Xcode'
+    inputs:
+      provisioning_script: $(Build.SourcesDirectory)/xamarin-macios/tools/devops/build-provisioning.csx
+      provisioning_extra_args: '-vvvv'
+      github-pat: $(GitHub.Token)
+    timeoutInMinutes: 250
+
+  # Disabled
+  - task: xamops.azdevex.provisionator-task.provisionator@1
+    displayName: 'Provision Objective Sharpie'
+    inputs:
+      provisioning_script: $(Build.SourcesDirectory)/xamarin-macios/tools/devops/provision-objective-sharpie.csx
+      provisioning_extra_args: '-vvvv'
+      github-pat: $(GitHub.Token)
+    enabled: false
+    timeoutInMinutes: 30
+
+  - task: xamops.azdevex.provisionator-task.provisionator@1
+    displayName: 'Provision .NET Core'
+    inputs:
+      provisioning_script: $(Build.SourcesDirectory)/xamarin-macios/tools/devops/provision-dotnetcore.csx
+      provisioning_extra_args: '-vvvv'
+      github-pat: $(GitHub.Token)
+    timeoutInMinutes: 30
+
+  - powershell: |
+      if ([IO.Directory]::Exists('/Developer/MonoTouch')) {
+        Write-Host "Deleting /Developer/MonoTouch"
+        bash -c "sudo rm -Rf /Developer/MonoTouch"
```

Comment:
You're embedding bash inside powershell? Seems like this could be simplified a lot by using bash directly.

Reply:
This is very dirty, i'll remove it @mcumming any reason to do it this way?